drivers/soundwire/mipi_disco.c

```diff
@@ -376,6 +376,12 @@ int sdw_slave_read_prop(struct sdw_slave *slave)
 	device_property_read_u32(dev, "mipi-sdw-sink-port-list",
 				 &prop->sink_ports);
 
+	device_property_read_u32(dev, "mipi-sdw-sdca-interrupt-register-list",
+				 &prop->sdca_interrupt_register_list);
+
+	device_property_read_u32(dev, "mipi-sdw-commit-register-supported",
+				 &prop->commit_register_supported);
```

Comment:
Hmm, `mipi-sdw-commit-register-supported` is a bool.

Reply:
yes, but the implementation of device_property_read_bool() isn't quite right, it only detects the presence of the property and not the value. In the MIPI DisCo spec, it's possible to have a property where 'commit-register-supported' is false.

Adding @andy-shev for more comments on how to deal with this odd configuration.

Reply:
> yes, but the implementation of device_property_read_bool() isn't quite right, it only detects the presence of the property and not the value.

It's the correct implementation, it's by design like that.  

> In the MIPI DisCo spec, it's possible to have a property where 'commit-register-supported' is false.

The bug is either in MIPI Disco Specification (if it's really boolean in accordance with _DSD specifications) **OR** the variable in the code should be integer (8-, 16-, 32- or 64-bits long).

Reply:
for my education, where is it specified that a boolean cannot have a 'false' value in the _DSD specs?

I read this in the dsd-guide.pdf page 7 https://github.com/UEFI/DSD-Guide/blob/main/dsd-guide.pdf

"
One use case could be a Boolean value ("yes" means the device has a read-only register that contains a serial number, "no" means the device has no such register)
"

But at the same time, I don't see a Boolean as a valid type in https://uefi.org/sites/default/files/resources/_DSD-device-properties-UUID.pdf

````
The Device Properties UUID
daffd814-6eba-4d8c-8a91-bc9bbf4aa301
defines the data format for the Package (Data Structure) immediately following it as a
list of Packages of length 2 (Properties) where the first element of each contained
Package (Key) must be a String and the second element of it (Value) must be:
• an Integer,
• a String,
• a Reference, or
• a Package consisting entirely of Integer, String, or Reference objects (and
specifically not containing a nested Package).
````

Wondering what the official guidance on Booleans might be?

Reply:
> for my education, where is it specified that a boolean cannot have a 'false' value in the _DSD specs?
> 
> I read this in the dsd-guide.pdf page 7 https://github.com/UEFI/DSD-Guide/blob/main/dsd-guide.pdf
> 
> ```
> One use case could be a Boolean value ("yes" means the device has a read-only register that contains a serial number, "no" means the device has no such register)
> ```

Oh, this makes a lot of confusion. They are talking about String value where "yes", "no" or other representations are possible to express boolean, but it's not really boolean in terms how it was designed (in the code / mind).

> But at the same time, I don't see a Boolean as a valid type in https://uefi.org/sites/default/files/resources/_DSD-device-properties-UUID.pdf

Of course, there is nothing like that in the ACPI. But we assume (to be compatible with other providers and unification of the API, see below) that boolean is the one which is simply defined by __presence__ of the property, meaning in ACPI it's represented as whatever (in the value) part.  Hence we may test the property:

    device_property_present(...); // gives true or false

and at the same time it means that if we treat the property as boolean, it's presence is enough. Meaning **each** property has (shadowed) boolean equal **true** associated with that.

> Wondering what the official guidance on Booleans might be?

Please, send a patch to that repository, it's indeed unclear at all. And it's definitely a documentation bug.

You also may come to OS Firmware meeting (next will be tomorrow) and talk about this issue. Might make sense to summon Sakari Ailus to the discussion.

Reply:
@andy-shev the DSDT of the device I have on my desk contains this:

````
Package (0x02)                                                            
{                                                                         
    "mipi-sdw-commit-register-supported",                                 
    Zero                                                                  
},
````

So at this point the cat is out of the bag.

Reply:
@plbossart It has to be handled as integer (let's say u8, but you need to see in the specification for the range of values) **if value matters**.

Reply:
used u8.